pypeit/find_objects.py

```diff
@@ -1010,26 +1009,103 @@ def find_objects_pypeline(self, image, ivar, std_trace=None,
+        if orig:
+            sobjs_ech = findobj_skymask.ech_objfind(
+                image, ivar, self.slitmask, self.slits_left, self.slits_right,
+                self.order_vec, self.reduce_bpm, det=self.det,
+                spec_min_max=np.vstack((self.slits.specmin, self.slits.specmax)),
+                inmask=inmask, 
+                ncoeff=self.par['reduce']['findobj']['trace_npoly'],
+                hand_extract_dict=manual_extract_dict, 
+                plate_scale=plate_scale,
+                std_trace=std_trace,
+                specobj_dict=specobj_dict,
+                snr_thresh=self.par['reduce']['findobj']['snr_thresh'],
+                show_peaks=show_peaks, show_fits=show_fits,
+                trim_edg=self.par['reduce']['findobj']['find_trim_edge'],
+                fwhm=self.par['reduce']['findobj']['find_fwhm'],
+                use_user_fwhm=self.par['reduce']['extraction']['use_user_fwhm'],
+                maxdev=self.par['reduce']['findobj']['find_maxdev'],
+                nperorder=nperorder,
+                max_snr=self.par['reduce']['findobj']['ech_find_max_snr'],
+                min_snr=self.par['reduce']['findobj']['ech_find_min_snr'],
+                nabove_min_snr=self.par['reduce']['findobj']['ech_find_nabove_min_snr'],
+                box_radius=self.par['reduce']['extraction']['boxcar_radius'],  # arcsec
+                show_trace=show_trace, objfindQA_filename=objfindQA_filename)
+        else:
+        '''
+
+        # Loop over the orders and find the objects within them (if any)
+        order_gpm = np.invert(self.reduce_bpm)
+        sobjs_in_orders = findobj_skymask.ech_findobj_ineach_order(
+            image, ivar, self.slitmask, self.slits_left, 
+            self.slits_right, self.slits.spat_id,
+            self.order_vec, order_gpm,
+            np.vstack((self.slits.specmin,  
+                        self.slits.specmax)),
+            plate_scale,
+            det=self.det,
+            inmask=inmask, 
             std_trace=std_trace,
             specobj_dict=specobj_dict,
             snr_thresh=self.par['reduce']['findobj']['snr_thresh'],
-            show_peaks=show_peaks, show_fits=show_fits,
+            show_peaks=show_peaks, 
             trim_edg=self.par['reduce']['findobj']['find_trim_edge'],
             fwhm=self.par['reduce']['findobj']['find_fwhm'],
             use_user_fwhm=self.par['reduce']['extraction']['use_user_fwhm'],
             nperorder=nperorder,
             maxdev=self.par['reduce']['findobj']['find_maxdev'],
-            max_snr=self.par['reduce']['findobj']['ech_find_max_snr'],
-            min_snr=self.par['reduce']['findobj']['ech_find_min_snr'],
-            nabove_min_snr=self.par['reduce']['findobj']['ech_find_nabove_min_snr'],
             box_radius=self.par['reduce']['extraction']['boxcar_radius'],  # arcsec
-            show_trace=show_trace, objfindQA_filename=objfindQA_filename)
+            objfindQA_filename=objfindQA_filename)
```

Comment:
Echelle object finding is a core functionality just like non-echelle object finding. The only object dependency is on specobjs. By adding the control flow that was in core here, one now needs all objects needed to instantiate find_objects, which adds more dependencies. Also, someone may just want to find objects but not sky-subtract, which this class does not allow one to do in a simple way. I don't see a current motivation for taking this control flow outside of core.